File: python/1684.Count-the-Number-of-Consistent-Strings.py
class Solution:
    def countConsistentStrings(self, allowed: str, words: List[str]) -> int:
        count = 0
        allowed_set = set(allowed)

        def isConsistentSet(word):
            for ch in word:
                if ch not in allowed_set:
                    return False
            
            return True

        for word in words:
            if isConsistentSet(word):
                count += 1

        return count

File: javascript/1684.Count-the-Number-of-Consistent-Strings.js
/**
 * @param {string} allowed
 * @param {string[]} words
 * @return {number}
 */
var countConsistentStrings = function (allowed, words) {
  let count = 0;
  const allowed_set = new Set(allowed);

  const isConsistentSet = (word) => {
    for (const ch of word) {
      if (!allowed_set.has(ch)) return false;
    }

    return true;
  };

  for (const word of words) {
    if (isConsistentSet(word)) {
      count += 1;
    }
  }

  return count;
};
